playwright/_impl/_connection.py

```diff
@@ -148,6 +162,12 @@ async def run_as_sync(self) -> None:
     async def run(self) -> None:
         self._loop = asyncio.get_running_loop()
         self._root_object = RootChannelOwner(self)
+        await self._transport.connect()
+
+        async def init() -> None:
+            self.playwright_future.set_result(await self._root_object.initialize())
+
+        self._loop.create_task(init())
```

Comment:
I don't think we can post tasks to after run due to run_as_sync, how does this work?

Reply:
not sure, but tests seem fine with it.